### DISCUSSION TITLE
Security policy in OPCUA

### DISCUSSION THREAD

**Discussion by vidyarao123** [FreeOpcUa/python-opcua#1244] [Q&A]
I have included all 3 levels of security in my code. None, Sign, Sign and Encrypt. 

server = Server()
url = "opc.tcp://127.0.0.1:4840"
server.set_endpoint(url)
security_policy = [ua.SecurityPolicyType.NoSecurity, ua.SecurityPolicyType.Basic256Sha256_SignAndEncrypt, ua.SecurityPolicyType.Basic256Sha256_Sign]
server.set_security_policy(security_policy)

But when I browse through UA Expert. Only none option displays. I am not able to see the other two security policies (Sign , Sign and Encrypt). Can somebody tell me if I am missing anything here. 

![UA Security level](https://user-images.githubusercontent.com/52528649/105945026-77682780-608a-11eb-803c-c28cea3c6d85.jpg)

**Comment by AndreasHeine** [comment]
Do you have a certificate and a privatekey for you application? u cant run sign or sign and encrypt without it!!!

**Reply by AndreasHeine** [reply]
https://github.com/FreeOpcUa/python-opcua/blob/master/examples/server-with-encryption.py

**but please dont use the example certificate and key in production!**

**Comment by vidyarao123** [comment]
Hi, 
sorry for my late reply. I do not have a certificate or a private key. Could you please tell me how do I generate one ?

I did search on google on how to generate these but it was not successful.

**Reply by AndreasHeine** [reply]
https://github.com/FreeOpcUa/python-opcua/blob/master/examples/generate_certificate.sh

you should study the examples and most opc ua related stuff you can find here: https://reference.opcfoundation.org/v104/

get the basics of opcua first (makes dev life easier)
